validator/src/consensus/signing/client.ts

```diff
@@ -55,13 +55,22 @@ export class SigningClient {
 		signatureId: SignatureId,
 		message: Hex,
 		sequence: bigint,
+		signers: readonly ParticipantId[],
 	): {
 		nonceCommitments: PublicNonceCommitments;
 		nonceProof: Hex[];
 	} {
-		// TODO: pass in signers to allow to remove unhonest
-		const participants = this.#storage.participants(groupId);
-		const signers = participants.map((p) => p.id).sort();
+		if (signers.length < this.#storage.threshold(groupId)) {
+			throw new Error("Not enough signers to start signing process");
```

Comment:
So, general comment (and doesn't have to be addressed here), but I don't think this is an error (i.e. an unexpected state). I think we need to handle this more gracefully and drop the signature request as opposed to failing the state transition. Especially if we want to recover for network outages.

This is in contrast to, for example, the "invalid signer id provided" error which should only happen because of a logic error (and therefore throwing an error makes sense).

Reply:
Yes, this should be handled in the state transition function and not the client.